### PR TITLE
develop: update connection page for TiDB Dedicated clusters

### PR DESCRIPTION
**Pull request by qqqdan**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [ ] master (the latest development version)
- [ ] v8.4 (TiDB 8.4 versions)
- [ ] v8.3 (TiDB 8.3 versions)
- [ ] v8.2 (TiDB 8.2 versions)
- [ ] v8.1 (TiDB 8.1 versions)
- [ ] v8.0 (TiDB 8.0 versions)
- [ ] v7.5 (TiDB 7.5 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: develop/dev-guide-sample-application-java-jdbc.md [hunk 1]
```diff
@@ -74,7 +74,7 @@ Connect to your TiDB cluster depending on the TiDB deployment option you've sele
 
 3. Ensure the configurations in the connection dialog match your operating environment.
 
-    - **Endpoint Type** is set to `Public`
+    - **Connection Type** is set to `Public`
```
Comment (Oreoxmt) [inline develop/dev-guide-sample-application-java-jdbc.md:77]:
```suggestion
    - **Endpoint Type** is set to `Public`
```
For TiDB Serverless clusters, it is "Endpoint Type" in the console.
image

**Comment by Oreoxmt** [comment]
Preview: https://pingcap-docsite-preview-git-preview-cl-01715d-oreoxmts-projects.vercel.app/tidbcloud/dev-guide-gui-datagrip

File: develop/dev-guide-sample-application-java-jdbc.md (hunk 1 above)
Comment (Oreoxmt) [inline develop/dev-guide-sample-application-java-jdbc.md:77]: https://github.com/pingcap/docs/pull/18756 renames **Endpoint Type** to **Connection Type** for TiDB Serverless clusters

**Review by Icemap — APPROVED** [review]
LGTM

**Review by lilin90 — APPROVED** [review]
/lgtm

**Review by Oreoxmt — APPROVED** [review]

**Pull request merged**